File: sortowanie_alfabetyczne.pas
program sortowanie_alfabetyczne; 
{ 
 --== Sortowanie alfabetyczne ==-- 
 Copyright(c)by Jan T. Biernat 
} 
uses crt; {CRT - biblioteka posiada funkcje/procedury 
                 do obslugi m.in. klawiatury i ekranu.} 
var   Tablica :array[0..14] of string; 
    I, AA, BB :Shortint; 
        tTemp :String; 
begin 
  TextBackground(Black); {Ustawienie czarnego koloru dla tla.} 
  TextColor(LightGray); {Ustawienie szarego koloru dla liter/czcionek.} 
  ClrScr; {Czyszczenie ekranu.} 
  Writeln('--== Sortowanie Alfabetyczne ==--'); 
  Writeln('Copyright (c)by Jan T. Biernat'); 
  Writeln; 
  {Wpisanie danych do tablicy.} 
  Tablica[0]:= 'Atari'; 
  Tablica[1]:= 'Amiga'; 
  Tablica[2]:= 'ZX Spectrum'; 
  Tablica[3]:= 'Amstrad'; 
  Tablica[4]:= 'Commodore'; 
  Tablica[5]:= 'Pegasus'; 
  Tablica[6]:= 'Atari Portfolio'; 
  Tablica[7]:= 'Atari TT'; 
  Tablica[8]:= 'Atari 65XE'; 
  Tablica[9]:= 'Joystick'; 
  Tablica[10]:= 'Rive Raid'; 
  Tablica[11]:= 'Robbo'; 
  Tablica[12]:= 'Fred'; 
  Tablica[13]:= 'Misja'; 
  Tablica[14]:= 'Lasermania'; 
  {Wyswietl dane.} 
  for I:= 0 to 14 do begin 
    GotoXY(4, 4+I); 
    Write(Tablica[I]); 
  end; 
  {Sortowanie alfabetyczne.} 
  for AA:= 0 to 14 do begin 
    for BB:= 0 to 14 do begin 
      if(Tablica[AA] < Tablica[BB]) then begin 
        tTemp:= ''; tTemp:= Tablica[BB]; 
        Tablica[BB]:= Tablica[AA]; 
        Tablica[AA]:= tTemp; 
      end; 
    end; 
  end; 
  {Wyswietl dane.} 
  for I:= 0 to 14 do begin 
    GotoXY(24, 4+I); 
    Write(Tablica[I]); 
  end; 
  {Nacisnij dowolny klawisz.} 
  Writeln; Writeln; Write('Nacisnij dowolny klawisz...'); 
  repeat until KeyPressed; {Petla wykonywana jest do czasu 
                            nacisniecia dowolnego klawisza 
                            przez uzytkownika.} 
end. 

File: sortowanie_alfabetyczne.cpp
#include <iostream> /*Biblioteka "iostream" (ang. input/output stream) 
                      oznacza strumień wejścia/wyjścia. 
                      Za pomocą tej biblioteki można m.in. wprowadzać 
                      informacje ze standardowych urządzeń 
                      wejścia (klawiatura) lub wyprowadzać 
                      informacje ze standardowych urządzeń 
                      wyjścia (ekran).*/ 
#include <conio.h>  /*Obsługa klawiatury za pomocą.*/ 
/*--== Sortowanie alfabetyczne ==-- 
  Copyright(c)by Jan T. Biernat*/ 
using namespace std; 
void TablicaJW_Pokaz(string Tablica[], int IloscElementow=0) { 
  //TablicaJW_Pokaz - Wyświetl zawartość tablicy podanej w 1 parametrze. 
    if(IloscElementow > 0) { 
      for(int I = 0; I < IloscElementow; I++) { 
        if(I > 0) { cout << ", "; } 
        cout << "\n" << Tablica[I]; 
      } 
      cout << "."; 
    } else { cout << "BLAD -?"; } 
} 
//Blok główny. 
int main() { 
  cout << "--== Sortowanie alfabetyczne ==--\n"; 
  cout << "Copyright (c)by Jan T. Biernat\n\n"; 
  //Deklaracja stałych. 
    const short int IloscElementow = 15; 
  //Deklaracja zmiennych. 
    string Tablica[IloscElementow]  = { 
                                        "Atari", 
                                        "Amiga", 
                                        "ZX Spectrum", 
                                        "Amstrad", 
                                        "Commodore", 
                                        "Pegasus", 
                                        "Atari Portfolio", 
                                        "Atari TT", 
                                        "Atari 65XE", 
                                        "Joystick", 
                                        "Rive Raid", 
                                        "Robbo", 
                                        "Fred", 
                                        "Misja", 
                                        "Lasermania" 
                                      }; 
    string Temp = ""; 
  //Wyświetl zawartość tablicy, przed posortowaniem alfabetycznym. 
    cout << "\nZawartosc tablicy, przed posortowaniem alfabetycznym:"; 
    TablicaJW_Pokaz(Tablica, IloscElementow); //Wywołanie funkcji "TablicaJW_Pokaz". 
    cout << "\n\n"; 
  //Sortowanie alfabetyczne. 
    for(int A = 0; A < IloscElementow; A++) { 
      for(int B = 0; B < IloscElementow; B++) { 
        if(Tablica[A] < Tablica[B]) { 
          Temp = ""; 
          Temp = Tablica[B]; 
          Tablica[B] = Tablica[A]; 
          Tablica[A] = Temp; 
        }; 
      }; 
    }; 
  //Wyświetl zawartość tablicy, po posortowaniu alfabetycznym. 
    cout << "\nZawartosc tablicy, po posortowaniu alfabetycznym:"; 
    TablicaJW_Pokaz(Tablica, IloscElementow); //Wywołanie funkcji "TablicaJW_Pokaz". 
    cout << "\n"; 
  //Naciśnij dowolny klawisz. 
    cout << "\n\nNacisnij dowolny klawisz..."; getch(); 
} 